### PR TITLE
Change timeout for  before/after commands to 300s, allow override via env var

### LINKED ISSUE #106: Allow before/after commands to run more than 60 seconds
Local npm/whatever builds can take a while, but the timeout is set to 60 seconds (default).

TODO: Increase the timeout to something reasonable: 180

### PR DESCRIPTION
**Pull request by tomdavies**
Fixes #106. 

I'm not convinced about using `craft\helpers\App::env()` here, it feels a bit wrong in this context (trait), but it also feels gross hardcoding a value, and `App::env()` is the Craft "way". 

I guess the other way to do it is pull from plugin settings / `config/copy.php`, but then we'd be accessing the plugin instance in the trait, which also doesn't feel very clean, but then I also don't use traits so much...

TLDR; Happy for this to be improved!

### CODE REVIEW

File: src/Helpers/DeployHooksHelper.php [hunk 1]
```diff
@@ -42,9 +44,12 @@ protected function runDeployCommands(string $when): bool
 
         $this->head("Run $when scripts", null, false);
 
+        $timeout = App::env('CRAFT_COPY_SCRIPT_TIMEOUT') ?: 300;
+
         foreach ($scripts as $script) {
             $this->cmdBlock($script);
             $process = Process::fromShellCommandline($script);
+            $process-setTimeout($timeout);
```
Comment (ostark) [inline src/Helpers/DeployHooksHelper.php:52]: ->

Comment (tomdavies) [reply src/Helpers/DeployHooksHelper.php]: idiocy fixed

**Comment by ostark** [comment]
I increased it to 1000s - hardcoded so far.

**Comment by Maybach91** [comment]
> [fe999b0](https://github.com/fortrabbit/craft-copy/commit/fe999b0387cf77043496f38b832e1b3bcbe30372)
> I increased it to 1000s - hardcoded so far.


@ostark :
I’m using craft-copy `1.1.0` (local and remote) and still get the error message for the 60 seconds timeout.
Also tried to set the `CRAFT_COPY_SCRIPT_TIMEOUT` env var to `1000`, but no luck.

```
[ERROR] The process "(cd buildchain && npm run prod)" exceeded the timeout of 60 seconds.                              
                                                                                               
```

**Comment by ostark** [comment]
Hey @Maybach91

this release should fix it eventually
https://github.com/fortrabbit/craft-copy/releases/tag/1.1.1
